app/views/nodes/modals/_add_node.html.erb

```diff
@@ -66,23 +69,24 @@
           # either because if we pass ':node' or ':nodes' form_for will try to
           # use the @node or @nodes instance variables and we get the same error. %>
         <%= form_tag(main_app.create_multiple_project_nodes_path(current_project), class: 'add_multiple_nodes_form') do %>
-          To create multiple nodes, add one node name per line:
 
           <div class="add_multiple_nodes_error">
             Please add at least one node
           </div>
 
           <%= hidden_field_tag "nodes[parent_id]", @node.id if type == :child %>
-          <%= text_area_tag    "nodes[list]", "", rows: 3, class: "form-control mb-3 nodes_list" %>
-
-          <%= label_tag :type_id, 'Icon' %>
+          <%= label_tag type == :child ? "node_#{@node.id}_nodes_list" : 'nodes_list', 'To create multiple nodes, add one node name per line:' %>
+          <%= text_area_tag "nodes[list]", "", rows: 3, class: 'form-control mb-3 nodes_list', id: type == :child ? "node_#{@node.id}_nodes_list" : 'nodes_list', data: { behavior: 'nodes-list' } %>
+          <%= label_tag type == :child ? "node_#{@node.id}_nodes_icon" : 'nodes_icon', 'Icon' %>
           <%= select_tag(
             "nodes[type_id]",
             options_for_select(
               [["No icon", Node::Types::DEFAULT], ["Host", Node::Types::HOST]]
             ),
             include_blank: false,
-            class: 'custom-select')
+            class: 'custom-select',
+            id: type == :child ? "node_#{@node.id}_nodes_icon" : 'nodes_icon',
```

Comment:
That would create a non-unique ID.

We are using `id: type == :child ? "node_#{@node.id}_icon" : 'node_icon'` in the from for adding a single node and here we are using `id: type == :child ? "node_#{@node.id}_nodes_icon" : 'nodes_icon'` for adding multiple nodes.

Looking at it now the single node select should have `id: type == :child ? "node_#{@node.id}_node_icon" : 'node_icon'` for consistency.

Reply:
The only difference in the caller is:
```bash
app//views/layouts/tylium.html.erb:    <%= render partial: 'nodes/modals/add_node', locals: { type: :branch } %>
app//views/layouts/tylium.html.erb:      <%= render partial: 'nodes/modals/add_node', locals: { type: :child } %>
```

Instead of adding a conditional everywhere you have an :id field, just prefix/suffix it w/ :type:

```ruby
           <%= label_tag "#{type}-nodes_list", 'To create multiple nodes, add one node name per line:' %> 
```

Reply:
I like it, I pushed this change. I had to prefix the hidden field in this case to prevent a unique id conflict.